app/controllers/responses_controller.rb

```diff
@@ -1,33 +1,44 @@
 class ResponsesController < ApplicationController
+  rescue_from BaseModel::RecordInvalid, with: :unprocessable_entity
+  rescue_from ActionController::ParameterMissing, with: :parameter_missing
+
   wrap_parameters Response, format: :json
 
   def create
     @response = Response.new(convert_response_params)
     render json: ResponseSerializer.new(@response).as_json
-  rescue ActionController::ParameterMissing
-    head :unprocessable_entity
   end
 
   private
 
   # TODO: update model to be equal to the model used by clientside so this conversion is unneccesary
   def convert_response_params
     {
-      answer_values: response_params['questions'].each_with_object({}) do |question, memo|
-        memo[question['key']] = question['answer_value']
-      end
+      answer_values: response_params[:questions].each_with_object({}) do |question, memo|
+        memo[question[:id]] = question[:answer_value]
+      end,
+      domain_ids: response_params[:domain_ids]
     }
   end
 
+  def parameter_missing(exception)
+    render json: { message: exception.message }, status: :unprocessable_entity
+  end
+
+  def unprocessable_entity(exception)
+    render json: { message: exception.message }, status: :unprocessable_entity
+  end
+
   def response_params
```

Comment:
Weet jij een manier om met strong parameters een *attribuut* required te maken? Ik niet. In pseudocode zou ik `params.require(:response).require(questions: [])` willen doen, maar dat mag niet van Rails?

Heb je suggestie over het omschrijven van de questions hash toegepast 👍